src/panels/config/energy/components/ha-energy-device-settings.ts

```diff
@@ -122,6 +143,21 @@ export class EnergyDeviceSettings extends LitElement {
     `;
   }
 
+  private _itemMoved(ev: CustomEvent): void {
+    ev.stopPropagation();
+    const { oldIndex, newIndex } = ev.detail;
+    const devices = this.preferences.device_consumption.concat();
+    const device = devices.splice(oldIndex, 1)[0];
+    devices.splice(newIndex, 0, device);
+
+    const newPrefs = {
+      ...this.preferences,
+      device_consumption: devices,
+    };
+    fireEvent(this, "value-changed", { value: newPrefs });
+    this._savePreferences(newPrefs);
```

Comment:
`_savePreferences` fires "value-changed" so this would fire the event twice

Reply:
This is true but somewhat intentional.

Note savePreferences is async which writes a save to the backend, and then fires valueChanged with the return value of that function ( I guess to reset if it has an error ).

But if I wait for that function to return when dropping the sortable, there is a visual glitch where the dropped item pops back to its original position for a few milliseconds (I guess for the duration of the async core save function). 

I did this so that it would lock in the new position of the sortable immediately and not have a visual glitch. Then when we get the final value back from savePrefs, we update the value again with that, in case it is something different.